src/main/java/uk/gov/justice/hmpps/prison/service/SentenceEnvelopeService.kt

```diff
@@ -0,0 +1,122 @@
+package uk.gov.justice.hmpps.prison.service
+
+import org.apache.commons.text.WordUtils
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.PageImpl
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Sort
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.hmpps.prison.api.model.BookingAdjustment
+import uk.gov.justice.hmpps.prison.api.model.FixedTermRecallDetails
+import uk.gov.justice.hmpps.prison.api.model.OffenderFinePaymentDto
+import uk.gov.justice.hmpps.prison.api.model.OffenderSentenceAndOffences
+import uk.gov.justice.hmpps.prison.api.model.SentenceAdjustmentValues
+import uk.gov.justice.hmpps.prison.api.model.calculation.CalculableSentenceEnvelope
+import uk.gov.justice.hmpps.prison.api.model.calculation.Person
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderAlert
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderBooking
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.AgencyLocationRepository
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderBookingId
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderBookingRepository
+import uk.gov.justice.hmpps.prison.service.transformers.OffenderAlertTransformer
+import java.util.Objects
+import java.util.stream.Collectors
+
+@Service
+@Transactional(readOnly = true)
+class SentenceEnvelopeService(
+  private val agencyLocationRepository: AgencyLocationRepository,
+  private val offenderFixedTermRecallService: OffenderFixedTermRecallService,
+  private val offenderBookingRepository: OffenderBookingRepository,
+  private val bookingService: BookingService,
+) {
+  fun getCalculableSentenceEnvelopeByEstablishment(
+    caseLoad: String,
+    pageNumber: Int,
+    pageSize: Int,
+  ): Page<CalculableSentenceEnvelope> {
+    val agencyLocation = agencyLocationRepository.getReferenceById(caseLoad)
+    val pageRequest = PageRequest.of(pageNumber, pageSize, Sort.by("bookingId"))
+    val bookingIds =
+      offenderBookingRepository.findDistinctByActiveTrueAndLocationAndSentences_statusAndSentences_CalculationType_CalculationTypeNotLikeAndSentences_CalculationType_CategoryNot(
+        agencyLocation,
+        "A",
+        "%AGG%",
+        "LICENCE",
+        pageRequest,
+      )
+    val activeBookings =
+      offenderBookingRepository.findAllByBookingIdIn(bookingIds.stream().map(OffenderBookingId::bookingId).toList())
```

Comment:
I've done a very simple conversion to kotlin - will fix the `.stream()` etc in next PR